compiler/semtempl.nim

```diff
@@ -620,12 +620,6 @@ proc semTemplateDef(c: PContext, n: PNode): PNode =
     s = semIdentVis(c, skTemplate, n[namePos], {})
   assert s.kind == skTemplate
 
-  if s.owner != nil:
-    const names = ["!=", ">=", ">", "incl", "excl", "in", "notin", "isnot"]
-    if sfSystemModule in s.owner.flags and s.name.s in names or
-       s.owner.name.s == "vm" and s.name.s == "stackTrace":
-      incl(s.flags, sfCallsite)
-
```

Comment:
What's the point of `sfCallsite` anyway?

Reply:
Like `{.line: instantiationInfo().}`, except applies to the top node of the template body, instead of recursively on nodes below it, meaning `.line` will wipe the info of template parameters and `callsite` doesn't really do anything to an indented template body. It's implemented for these procs because `instantiationInfo` is lower in system than where they're implemented (it has to be below `arithmetics` because it uses `-1`, but it has to be above `comparisons` which is the next line after arithmetics).

I can sandwich `instantiationInfo` between arithmetics and comparisons, or I can try to make sfCallsite recursively copy the line info while not modifying parameter info (which it should be able to do, since it's accounted for before the template is evaluated). I'm partial towards improving callsite because it's more terse than `.line`, maybe we can reuse `ccInline` with `.inline` to save a symbol flag spot.